File: RailgunVB/Core/SystemBase.vb
Imports System.IO
Imports System.Text
Imports Discord
Imports Discord.Commands

Namespace Core

    Public Class SystemBase
        Inherits ModuleBase(Of SystemContext)

        Protected Overrides Sub AfterExecute(command As CommandInfo)
            Context.DisposeDatabase()
            MyBase.AfterExecute(command)
        End Sub
        
        Protected Async Function SendStringAsFileAsync(tc As ITextChannel, name As String, output As String, 
                                                    Optional text As String = Nothing, 
                                                    Optional includeGuildName As Boolean = True) As Task
            Await tc.SendFileAsync(New MemoryStream(Encoding.UTF8.GetBytes(output)), 
                                   $"{If(includeGuildName, $"{tc.Guild.Name}-", "") + name}", text)
        End Function
        
    End Class
End NameSpace

File: Railgun/Core/Commands/SystemBase.cs
using System.Threading.Tasks;
using Discord;
using Finite.Commands;
using Railgun.Core.Extensions;

namespace Railgun.Core.Commands
{
    public class SystemBase : ModuleBase<SystemContext> { 
        public async Task<IUserMessage> ReplyAsync(string msg = null, bool isTTS = false, Embed embed = null) 
            => await (Context.Channel as ITextChannel).TrySendMessageAsync(msg, isTTS, embed);
    }
}